src/main/java/com/sforce/cd/apexUnit/client/connection/ConnectionHandler.java

```diff
@@ -106,21 +115,16 @@ public static ConnectionHandler getConnectionHandlerInstance() {
 			
 	private PartnerConnection createConnection() {
 		if (connection == null) {
-			ConnectorConfig config = new ConnectorConfig();
-			config.setUsername(CommandLineArguments.getUsername());
-			config.setPassword(CommandLineArguments.getPassword());
+			ConnectorConfig config = createConfig();
 			config.setAuthEndpoint(CommandLineArguments.getOrgUrl() + "/services/Soap/u/" + SUPPORTED_VERSION);
-			
-			if (CommandLineArguments.getProxyHost() != null &&  CommandLineArguments.getProxyPort() !=null ){
-				LOG.debug("Setting proxy configuraiton to " + 
-					CommandLineArguments.getProxyHost() + 
-					" on port " + CommandLineArguments.getProxyPort() );
-				config.setProxy( CommandLineArguments.getProxyHost() , CommandLineArguments.getProxyPort());
-			}	
-			
+			config.setSessionRenewer(new SFDCSessionRenewer());
+			LOG.info("Default connection time out value is: " + config.getConnectionTimeout());
+			config.setConnectionTimeout(MAX_TIME_OUT_IN_MS_INT);
```

Comment:
per your initial analysis, it appeared that this was zero. Thoughts on why you see a value in customizing? please help me recall the decision

Is this a mandatory field? If not, you should check for -ve or null values before this set